File: 466. 链表节点计数.cpp
/**
 * Definition of singly-linked-list:
 * class ListNode {
 * public:
 *     int val;
 *     ListNode *next;
 *     ListNode(int val) {
 *        this->val = val;
 *        this->next = NULL;
 *     }
 * }
 */

class Solution {
public:
    /**
     * @param head: the first node of linked list.
     * @return: An integer
     */
    int countNodes(ListNode * head) {
        int size = 0;
        while (head != nullptr) {
            size += 1;
            head = head->next;
        }
        return size;
    }
};

// 总耗时 89 ms
// 您的提交打败了 98.40% 的提交!

File: Python3/466. 链表节点计数.py
"""
Definition of ListNode
class ListNode(object):
    def __init__(self, val, next=None):
        self.val = val
        self.next = next
"""

class Solution:
    """
    @param head: the first node of linked list.
    @return: An integer
    """
    def countNodes(self, head):
        length = 0
        while head :
            length += 1
            head = head.next
        return length

# 总耗时 1308 ms
# 您的提交打败了 69.30% 的提交!